src/main/java/org/jabref/logic/openoffice/action/EditMerge.java

```diff
@@ -0,0 +1,344 @@
+                return false;
+            }
+
+            if (!UnoTextRange.comparables(state.prevRange, currentRange)) {
+                return false;
+            }
+
+            // Sanity check: the current range should start later than the previous.
+            int textOrder = UnoTextRange.compareStarts(state.prevRange, currentRange);
+            if (textOrder != (-1)) {
+                String msg =
+                    String.format("MergeCitationGroups:"
+                                  + " \"%s\" supposed to be followed by \"%s\","
+                                  + " but %s",
+                                  state.prevRange.getString(),
+                                  currentRange.getString(),
+                                  ((textOrder == 0)
+                                   ? "they start at the same position"
+                                   : "the start of the latter precedes the start of the first"));
+                LOGGER.warn(msg);
+                return false;
+            }
+        }
+
+        if (state.cursorBetween == null) {
+            return false;
+        }
+
+        Objects.requireNonNull(state.cursorBetween);
+        Objects.requireNonNull(state.currentGroupCursor);
+
+        // assume: currentGroupCursor.getEnd() == cursorBetween.getEnd()
+        if (UnoTextRange.compareEnds(state.cursorBetween, state.currentGroupCursor) != 0) {
+            String msg = ("MergeCitationGroups: cursorBetween.end != currentGroupCursor.end");
+            throw new IllegalStateException(msg);
+        }
+
+        /*
+         * Try to expand state.currentGroupCursor and state.cursorBetween by going right to reach
+         * rangeStart.
+         */
+        XTextRange rangeStart = currentRange.getStart();
+        boolean couldExpand = true;
+        XTextCursor thisCharCursor =
+            (currentRange.getText().createTextCursorByRange(state.cursorBetween.getEnd()));
+
+        while (couldExpand && (UnoTextRange.compareEnds(state.cursorBetween, rangeStart) < 0)) {
+            //
+            // Check that we only walk through inline whitespace.
+            //
+            couldExpand = thisCharCursor.goRight((short) 1, true);
+            String thisChar = thisCharCursor.getString();
+            thisCharCursor.collapseToEnd();
+            if (thisChar.isEmpty() || "\n".equals(thisChar) || !thisChar.trim().isEmpty()) {
+                couldExpand = false;
+                if (!thisChar.isEmpty()) {
+                    thisCharCursor.goLeft((short) 1, false);
+                }
+                break;
+            }
+            state.cursorBetween.goRight((short) 1, true);
+            state.currentGroupCursor.goRight((short) 1, true);
+
+            // These two should move in sync:
+            if (UnoTextRange.compareEnds(state.cursorBetween, state.currentGroupCursor) != 0) {
+                String msg = ("MergeCitationGroups: cursorBetween.end != currentGroupCursor.end (during expand)");
```

Comment:
Can be inlined.
Please avoid technical details in error messages, thats up for debug or logger. End user must be able to understand why this fails.

Reply:
changed to
```
                LOGGER.warn("MergeCitationGroups: cursorBetween.end != currentGroupCursor.end (during expand)");
                throw new IllegalStateException("MergeCitationGroups failed");
```